### PR TITLE
How-To: k8s pod patching

### PR DESCRIPTION
**Pull request by willtsai**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

add how to guide for k8s pod patching

### Summary
📝🛠️🐳

Add a new guide on patching Kubernetes resources with PodSpec. The guide explains how to create, deploy, and test an app and a container with an ephemeral volume using code snippets, screenshots, and links.

> _Learn to patch `PodSpec`_
> _Customize containers, volumes_
> _A guide for autumn_

### Walkthrough
* Add a new how-to guide on patching Kubernetes resources using PodSpec definitions ([link](https://github.com/radius-project/docs/pull/800/files?diff=unified&w=0#diff-e9494ae71d9eb6464a4a551c552ca86dd2a37da57cae6dfa8fd25eef0d4bd142R1-R79))

## Issue reference

Fixes: #526

### CODE REVIEW

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+1. Deploy your app with the command:
+
+   ```bash
+   rad deploy ./app.bicep
+   ```
+1. Once complete, port forward to your container with [`rad resource expose`]({{< ref rad_resource_expose >}}):
+
+   ```bash
+   rad resource expose containers mycontainer -a myapp --port 5000
+   ```
```
Comment (rynowak) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md:71]: Can we use `rad run` instead?

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+description: "Learn how to patch Kubernetes resources using PodSpec definitions"
+weight: 400
+slug: 'patch-podspec'
+categories: "How-To"
+tags: ["containers","Kubernetes"]
+---
+
+This how-to guide will provide an overview of how to:
+
+- Mount an ephemeral (short-lived) volume to a container
```
Comment (rynowak) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: What makes it short-lived?

Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: PR still in draft, this was copied from another how to that I just haven't removed yet :)

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md (hunk 1 above)
Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md:71]: Yes, I plan to do that - this was copied from another how-to, I haven't pushed my latest yet. Still in draft, please do not review again until I mark it as ready!

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+   demo-df76d886c-sngm8 demo [port-forward] connected from localhost:3000 -> ::3000
+   ```
+
+   Access the application by opening [http://localhost:3000](http://localhost:3000) in a browser, where you should see the demo app:
+
+   <img src="./demoapp-howtopatchpod.png" alt="Screenshot of Radius Demo app" width="600"/>
+
+   <br> When you're ready to move on to the next step, use `CTRL` + `C` to exit the command.
+
+2. Run the following command, making sure to specify the Kubernetes namespace to which you deployed your app:
```
Comment (rynowak) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: How will the user figure this out? I'd suggest adding an `-A` to the command and then they won't need to know.

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+   ```bash
+   kubectl get pods -n <YOUR_KUBERNETES_NAMESPACE> -l app.kubernetes.io/name=demo -o custom-columns=POD:.metadata.name,STATUS:.status.phase,CONTAINER_NAMES:spec.containers[:].name,CONTAINER_IMAGES:spec.containers[:].image
+   ```
+
+   You should now see in the output the original `demo` app container as before, but also an additional `log-collector` container that is running in your pod, similar to the following:
+
+   ```
+   POD                     STATUS    CONTAINER_NAMES      CONTAINER_IMAGES
+   demo-547d7dc77f-nmqpk   Running   log-collector,demo   ghcr.io/radius-project/fluent-bit:2.1.8,radius.azurecr.io/tutorial/webapp:edge
+   ```
```
Comment (rynowak) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md:120]: This is a really cool way of demonstrating the effect. 

Depending on the speed the user does this, they might see old pods with a state of `Terminating`. You might want to mention that, and say that it's normal.

**Review by rynowak — COMMENTED** [review]
I left some very minor suggestions. It think this is in really good shape overall.

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+## Prerequisites
+
+- [rad CLI]({{< ref getting-started >}})
+- [Radius initialized with `rad init`]({{< ref howto-environment >}})
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Step 1: Define an app and a container
+
+Begin by creating a file named `app.bicep` with a Radius application and [container]({{< ref "guides/author-apps/containers" >}}):
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]:
```suggestion
## Step 1: Define a container

Begin by creating a file named `app.bicep` with a Radius [container]({{< ref "guides/author-apps/containers" >}}):
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+{{< rad file="snippets/patch-container.bicep" embed=true >}}
+
+## Step 2: Deploy the app and container
+
+1. Deploy and run your app:
+
+   ```bash
+   rad run ./app.bicep
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: Without an explicit "Run rad init in a new directory and specify 'y' we'll need to pass in the application at run:
```suggestion
   rad run ./app.bicep -a demo
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md (hunk 3 above)
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: Can we also add a quick explanation about what the `-o` flag does. Maybe "Run the following command to get the running pods and their name, status, container names, and image names."

User without much kubectl knowledge may see a bunch of commands and think they're important to read and not just a display optimmization.

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+   ```bash
+   kubectl get pods -n <YOUR_KUBERNETES_NAMESPACE> -l app.kubernetes.io/name=demo -o custom-columns=POD:.metadata.name,STATUS:.status.phase,CONTAINER_NAMES:spec.containers[:].name,CONTAINER_IMAGES:spec.containers[:].image
+   ```
+
+   You should see output confirming that a single container named `demo` was deployed and is running in your pod, similar to the following:
+
+   ```
+   POD                    STATUS    CONTAINER_NAMES   CONTAINER_IMAGES
+   demo-df76d886c-9p4gv   Running   demo              radius.azurecr.io/tutorial/webapp:edge
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: Pls use the new GHCR image

```suggestion
   demo-df76d886c-9p4gv   Running   demo              ghcr.io/radius-project/samples/demo:latest
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+{{< rad file="snippets/patch-runtime.bicep" embed=true markdownConfig="{linenos=table,hl_lines=[\"16-28\"]}" >}}
+
+> Remember to save your `app.bicep` file after you've made the above changes.
+
+## Step 4: Redeploy your app with the PodSpec added
+
+1. Deploy and run your app again:
+
+   ```bash
+   rad run ./app.bicep
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]:
```suggestion
   rad run ./app.bicep -a demo
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+   ```
+
+   The `log-collector` container was deployed using the PodSpec definition you added to your `app.bicep` file in the `runtimes` property you added, and is now running alongside your original `demo` app container.
+
+## Cleanup
+
+Run the following command to [delete]({{< ref "guides/deploy-apps/howto-delete" >}}) your app and container:
+   
+   ```bash
+   rad app delete
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]:
```suggestion
   rad app delete demo
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/snippets/patch-runtime.bicep [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+@description('The ID of your Radius Application. Automatically injected by the rad CLI.')
+param application string
+
+resource demo 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'radius.azurecr.io/tutorial/webapp:edge'
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/snippets/patch-runtime.bicep]:
```suggestion
      image: 'ghcr.io/radius-project/samples/demo:latest'
```

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+   You should see output confirming that a single container named `demo` was deployed and is running in your pod, similar to the following:
+
+   ```
+   POD                    STATUS    CONTAINER_NAMES   CONTAINER_IMAGES
+   demo-df76d886c-9p4gv   Running   demo              radius.azurecr.io/tutorial/webapp:edge
+   ```
+
+## Step 3: Add a PodSpec to the container definition
+
+Add the following [`runtimes`]({{< ref "reference/resource-schema/core-schema/container-schema#runtimes" >}}) under `properties` of the container definition in your `app.bicep` file:
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: I'd reword this slightly to talk about what this section is a little more. Don't want this to seem like a strongly typed property in Radius, but a way to access any part of the podspec and punch through the abstraction.

```suggestion
Add the following [`runtimes`]({{< ref "reference/resource-schema/core-schema/container-schema#runtimes" >}}) configuration to the container definition in your `app.bicep` file. This allows you to punch through the Radius abstraction and directly apply any part of the Kubernetes PodSpec. In this example you're adding an additional sidecar container:
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md (hunk 11 above)
Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: good suggestion, edited slightly

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md (hunk 3 above)
Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: I thought about this, but was afraid they'd have a bunch of namespaces and `-a` would output too large of a list. But I suppose that's unlikely given that one of the prereqs is to set up a cluster to play around with, so they might start with something that's fairly fresh. 

**tl;dr is that I'll add it :)**

Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md]: I assumed that most would have some knowledge of kubectl (or at least k8s) given the use case, but I agree `-o` is obscure enough to warrant more description

File: docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md (hunk 4 above)
Comment (willtsai) [reply docs/content/guides/author-apps/kubernetes/how-to-patch-pod/index.md:120]: good suggestion, done

**Review by AaronCrawfis — APPROVED** [review]
